lib/src/commands/Deprecations.ts

```diff
@@ -11,10 +11,19 @@ export class Deprecations {
     ) {
       this.deprecateBottomTabsVisibility(parentOptions);
     }
+    if (key === 'searchBar' && Platform.OS === 'ios' && typeof parentOptions[key] === 'boolean') {
+      this.deprecateSearchBarOptions(parentOptions);
+    }
   }
 
   public onProcessDefaultOptions(_key: string, _parentOptions: Record<string, any>) {}
 
+  private deprecateSearchBarOptions = once((parentOptions: object) => {
```

Comment:
@mateioprea Seems like all current `SearchBar` options need to be deprecated: 
```
  searchBar?: boolean;
  searchBarHiddenWhenScrolling?: boolean;
  searchBarPlaceholder?: string;
  searchBarBackgroundColor?: string;
  searchBarTintColor?: string;
  hideNavBarOnFocusSearchBar?: boolean;
```

We should do the following:
1. Warn when a deprecated option is used
2. Options processor should be able to process both old and new api, where the new API takes precedence over the old.
3. Native should only handle the new API